src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -686,10 +686,17 @@ private static Type GetTargetType(ApiDescription description, ApiParameterDescri
         var parameterType = parameter.Type is not null
             ? Nullable.GetUnderlyingType(parameter.Type) ?? parameter.Type
             : parameter.Type;
-        var requiresModelMetadataFallback = parameterType == typeof(string) && parameter.ModelMetadata.ModelType != parameter.Type;
+
+        // parameter.Type = typeof(string)
+        // parameter.ModelMetadata.Type = typeof(TEnum)
+        var requiresModelMetadataFallbackForEnum = parameterType == typeof(string)
+            && parameter.ModelMetadata.ModelType != parameter.Type
+            && parameter.ModelMetadata.ModelType.IsEnum;
         // Enums are exempt because we want to set the OpenApiSchema.Enum field when feasible.
+        // parameter.Type = typeof(TEnum), typeof(Student), typeof(ITryParse)
```

Comment:
`typeof(Student)` won't mean anything when we forget this issue in a few months 😄 
`typeof(typeWithTryParse)`?

Reply:
```suggestion
        // parameter.Type = typeof(TEnum), typeof(TypeWithTryParse)
```